### PR TITLE
bug: Improve chassis_types and security_breach columns within chassis_info

### PR DESCRIPTION
**Pull request by farfella**
Fix for bug #6538
* Correctly returns chassis type if only one chassis type.
* Correctly returns multiple chassis types.
* Table schema has been updated so that this column is now text and not an integer.

### CODE REVIEW

File: osquery/core/windows/wmi.h [hunk 1]
```diff
@@ -205,6 +205,16 @@ class WmiResultItem {
   Status GetVectorOfStrings(const std::string& name,
                             std::vector<std::string>& ret) const;
 
+  /**
+   * @brief Windows WMI Helper function to retrieve a vector of long result
+   * from
```
Comment (theopolis) [inline osquery/core/windows/wmi.h:210]: nitpick, formatting, no need to have a newline here

File: osquery/tables/system/windows/chassis_info.cpp [hunk 2]
```diff
@@ -33,11 +33,21 @@ QueryData genChassisInfo(QueryContext& context) {
...
     r["audible_alarm"] = isPresent ? "True" : "False";
     data.GetString("BreachDescription", r["breach_description"]);
-    data.GetLong("ChassisTypes", number);
-    r["chassis_types"] = INTEGER(number);
+    data.GetVectorOfLongs("ChassisTypes", chassisTypes);
+
+    std::ostringstream oValueConcat;
+    for (size_t i = 0; i < chassisTypes.size(); ++i) {
+      if (i != 0) {
+        oValueConcat << ",";
```
Comment (theopolis) [inline osquery/tables/system/windows/chassis_info.cpp:45]: nitpick, do you want this to read `1,2,3,4` or `1, 2, 3, 4`? If you do not want the space maybe you can `<< ','`.

File: specs/windows/chassis_info.table [hunk 3]
```diff
@@ -3,7 +3,7 @@ description("Display information pertaining to the chassis and its security stat
 schema([
     Column("audible_alarm", TEXT, "If TRUE, the frame is equipped with an audible alarm."),
     Column("breach_description", TEXT, "If provided, gives a more detailed description of a detected security breach."),
-    Column("chassis_types", INTEGER, "The type of chassis, such as Desktop or Laptop."),
+    Column("chassis_types", TEXT, "The type of chassis, such as Desktop or Laptop."),
```
Comment (theopolis) [inline specs/windows/chassis_info.table:6]: A comma-separated list of chassis types, such as Desktop or Laptop.

**Comment by farfella** [comment]
Thanks for the review @theopolis ! I updated the data in the column so `chassis_types` now has readable values, e.g., Laptop, vs integral values.

Also, noticed that `security_status` column was actually presenting values from SecurityBreach field (and not the separate SecurityStatus field) so I updated this to `security_breach` now. This is an unsigned short field instead of a long (ref: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-systemenclosure), and it may in fact be non-existent. So, there's now a check to return Unknown (2) if it is non-existent instead of returning garbage as what was happening before.

**Comment by theopolis** [comment]
This is great!

Heads up that I normally recommend a "one thesis rule", meaning each PR should change one distinct thing. I renamed this to indicate that you are essentially improving the data within both of these columns.

**Comment by farfella** [comment]
I agree with the one thesis rule. :) Integration test was failing because result of security_breach was uninitialized sometimes.

**Pull request merged**